engine/classes/Elgg/Database/EntityTable.php

```diff
@@ -1235,4 +1265,31 @@ function updateLastAction($guid, $posted = null) {
 			return false;
 		}
 	}
-}
\ No newline at end of file
+
+	/**
+	 * Get a user by GUID even if the entity is hidden or disabled
+	 *
+	 * @param int $guid User GUID. Default is logged in user
+	 *
+	 * @return \ElggUser|false
+	 * @throws UserFetchFailureException
+	 * @access private
+	 */
+	public function getUserForPermissionsCheck($guid = 0) {
+		if (!$guid) {
+			return _elgg_services()->session->getLoggedInUser();
```

Comment:
Anyway, thanks to this function we now have a good place to trigger the deprecation message, in case we later decide to deprecate the magic.